modules/api/pkg/resources/machine/machinedeployment.go

```diff
@@ -47,7 +50,12 @@ const (
 )
 
 // Deployment returns a Machine Deployment object for the given Node Deployment spec.
-func Deployment(c *kubermaticv1.Cluster, nd *apiv1.NodeDeployment, dc *kubermaticv1.Datacenter, keys []*kubermaticv1.UserSSHKey) (*clusterv1alpha1.MachineDeployment, error) {
+func Deployment(ctx context.Context, c *kubermaticv1.Cluster, nd *apiv1.NodeDeployment, dc *kubermaticv1.Datacenter, keys []*kubermaticv1.UserSSHKey, settingsProvider provider.SettingsProvider) (*clusterv1alpha1.MachineDeployment, error) {
+	validationErr := ValidateAutoUpdateMDEnforcement(ctx, nd, settingsProvider)
```

Comment:
I added this here because this covers all scenarios like Add/Edit MD, Create Cluster and Create/Update Cluster Template. However, this required adding some extra parameters and passing them through multiple layers of method calls.